### PR TITLE
Lint keyboard name against the requirements within the documentation

### PR DESCRIPTION
**Pull request by zvecr**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Will catch examples like 10980

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: lib/python/qmk/cli/lint.py [hunk 1]
```diff
@@ -46,6 +48,11 @@ def lint(cli):
         ok = False
         cli.log.error('Missing %s', readme_path)
 
+    # Check keyboard name is valid
+    if not re.fullmatch(r"[a-z0-9][a-z0-9_]*", cli.config.lint.keyboard):
+        ok = False
+        cli.log.error('Invalid keyboard name')
```
Comment (Erovia) [inline lib/python/qmk/cli/lint.py:54]:
```suggestion
        cli.log.error('Invalid keyboard name! Guidelines: https://docs.qmk.fm/#/hardware_keyboard_guidelines?id=naming-your-keyboardproject')
```
Providing the link in the output might be helpful.

**Review by Erovia — APPROVED** [review]

File: lib/python/qmk/cli/lint.py [hunk 2]
```diff
@@ -46,6 +48,11 @@ def lint(cli):
         ok = False
         cli.log.error('Missing %s', readme_path)
 
+    # Check keyboard name is valid
+    if not re.fullmatch(r"[a-z0-9][a-z0-9_]*", cli.config.lint.keyboard):
```
Comment (noroadsleft) [inline lib/python/qmk/cli/lint.py:52]: This regex doesn't match keyboards that use subdirectories... like... your own: :laughing: 

```
$ qmk lint -kb zvecr/zv48
ERROR Invalid keyboard name
ERROR Lint check failed!
```

```suggestion
    if not re.fullmatch(r"[a-z0-9][a-z0-9_/]*", cli.config.lint.keyboard):
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: lib/python/qmk/cli/lint.py (hunk 2 above)
Comment (zvecr) [reply lib/python/qmk/cli/lint.py:52]: Good catch!

File: lib/python/qmk/cli/lint.py [hunk 3]
```diff
@@ -46,6 +48,11 @@ def lint(cli):
         ok = False
         cli.log.error('Missing %s', readme_path)
 
+    # Check keyboard name is valid
+    if not re.fullmatch(r"[a-z0-9][a-z0-9_/]*", cli.config.lint.keyboard):
```
Comment (skullydazed) [inline lib/python/qmk/cli/lint.py:52]: This doesn't actually match our requirements.

We probably want to iterate over each path component and use a check function that looks something like this:

```
from string import ascii_lowercase, digits

first_letter_ok = ascii_lowercase + digits
rest_ok = first_letter_ok + '_'

def is_kb_name(path_component):
    if path_component[0] not in first_letter_ok:
        return False

    for letter in path_component[1]:
        if letter not in rest_ok:
            return False

    return True
```

Comment (zvecr) [reply lib/python/qmk/cli/lint.py]: Nice catch! Does something like the recent update look okay now?

File: lib/python/qmk/cli/lint.py (hunk 1 above)
Comment (zvecr) [reply lib/python/qmk/cli/lint.py:63]: This isn't a pattern that's used outside doctor right now. For example, `info.json` doesn't point to <https://docs.qmk.fm/#/reference_info_json>. Is this a general strategy we want to move towards?

**Comment by zvecr** [comment]
No review, CBA to take this forward so it can become someone elses problem

**Pull request closed without merging**